homeassistant/components/openhome/media_player.py

```diff
@@ -19,34 +20,81 @@
     SUPPORT_VOLUME_SET,
     SUPPORT_VOLUME_STEP,
 )
-from homeassistant.const import STATE_IDLE, STATE_OFF, STATE_PAUSED, STATE_PLAYING
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    STATE_IDLE,
+    STATE_OFF,
+    STATE_PAUSED,
+    STATE_PLAYING,
+)
+import homeassistant.helpers.config_validation as cv
+
+from .const import ATTR_PIN_INDEX, DATA_OPENHOME, DOMAIN, SERVICE_INVOKE_PIN
+
+OPENHOME_PIN_SCHEMA = vol.Schema(
+    {
+        vol.Required(ATTR_ENTITY_ID): cv.entity_id,
+        vol.Required(ATTR_PIN_INDEX): cv.positive_int,
+    }
+)
 
 SUPPORT_OPENHOME = SUPPORT_SELECT_SOURCE | SUPPORT_TURN_OFF | SUPPORT_TURN_ON
 
 _LOGGER = logging.getLogger(__name__)
 
-DEVICES = []
-
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the Openhome platform."""
 
     if not discovery_info:
         return True
 
+    if DATA_OPENHOME not in hass.data:
+        hass.data[DATA_OPENHOME] = []
+
     name = discovery_info.get("name")
     description = discovery_info.get("ssdp_description")
+
     _LOGGER.info("Openhome device found: %s", name)
     device = Device(description)
```

Comment:
Since this does a request we need to schedule the instantiation on the executor thread pool with `hass.async_add_executor_job`.

```suggestion
    device = await hass.async_add_executor_job(Device, description)
```